lib/matplotlib/contour.py

```diff
@@ -857,10 +857,12 @@ def __init__(self, ax, *args, **kwargs):
         else:
             self.logscale = False
 
-        if self.origin is not None:
-            assert(self.origin in ['lower', 'upper', 'image'])
-        if self.extent is not None:
-            assert(len(self.extent) == 4)
+        if self.origin not in [None, 'lower', 'upper', 'image']:
+            raise ValueError("If given, *origin* must be one of ['lower', \
+                    'upper', 'image']")
+        if self.extent is not None and len(self.extent) != 4:
+            raise ValueError("If give, *extent* must be '[ *None* | \
```

Comment:
you mean "implied string continuation". I hate seeing people use unneeded
plus signs for that.

On Sun, Mar 15, 2015 at 9:40 PM, Thomas A Caswell <notifications@github.com>
wrote:

> In lib/matplotlib/contour.py
> <https://github.com/matplotlib/matplotlib/pull/3060#discussion_r26455813>:
>
> > @@ -857,10 +857,12 @@ def __init__(self, ax, *args, **kwargs):
> >          else:
> >              self.logscale = False
> >
> > -        if self.origin is not None:
> > -            assert(self.origin in ['lower', 'upper', 'image'])
> > -        if self.extent is not None:
> > -            assert(len(self.extent) == 4)
> > +        if self.origin not in [None, 'lower', 'upper', 'image']:
> > +            raise ValueError("If given, *origin* must be one of ['lower', \
> > +                    'upper', 'image']")
> > +        if self.extent is not None and len(self.extent) != 4:
> > +            raise ValueError("If give, *extent* must be '[ *None* | \
>
> don't need \, use string concatenation please..
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/3060/files#r26455813>.
>